test-data/unit/check-enum.test

```diff
@@ -59,6 +59,26 @@ reveal_type(Truth.true.name)    # N: Revealed type is 'Literal['true']?'
 reveal_type(Truth.false.value)  # N: Revealed type is 'builtins.bool'
 [builtins fixtures/bool.pyi]
 
+[case testEnumValueExtended]
+from enum import Enum
+class Truth(Enum):
+    true = True
+    false = False
+
+def infer_truth(truth: Truth) -> None:
+    reveal_type(truth.value) # N: Revealed type is 'builtins.bool'
+[builtins fixtures/bool.pyi]
+
+[case testEnumValueInhomogenous]
+from enum import Enum
+class Truth(Enum):
+    true = 'True'
+    false = 0
+
+def cannot_infer_truth(truth: Truth) -> None:
+    reveal_type(truth.value) # N: Revealed type is 'Any'
```

Comment:
I wonder if this shouldn't infer `object` instead of `Any`?

In favor of `Any`: That's what we did before in this case; it prevents false positives.

In favor of `object`: That's what you'd see for e.g. `reveal_type("" if a else 0)`, assuming `a` has an unknown value.

Reply:
I was trying to just follow the existing art.  I imagine that changing to `object` could cause some code to fail with false positives (If a user is actually _using_ the `.value`, it seems unlikely that they will be restricting themselves to the interface provided by `object` so this would probably force them to do additional `cast` or other methods of type-narrowing)

With that said ... once this is implemented, it shouldn't matter to me either way.  Hopefully `mypy` will know the types of all of my enum values since I see no reason for inhomogenous values and it should never bother me in either case :).  I'll do whatever you like.

Reply:
Okay, let's stick with tradition and keep `Any`.